### PR TITLE
Extract dynamic_form's error_messages_for to avoid a wrong i18n scope 

### LINKED ISSUE #1834: Translation missing for pt-BR
Hey folks 👋 

When the user submit the register form for pt-BR, it returns a translation missing error:

image

It looks like that it isn't loading the translations for the https://github.com/svenfuchs/rails-i18n/ gem. I'm trying to fix it in the code but had no success yet. Any ideas? 🤔

### PR DESCRIPTION
**Pull request by duduribeiro**
Closes #1834

hey folks 👋 , I don't know if this is valid, but follow the PR and a description of the problem. 

Rubygems currently explodes with a translation missing error when submitting a wrong form.

image

The problem occurs because we use `form.error_messages` in the form and
dynamic_form adds `activerecord.errors.template.{header/body}` i18n key
when it has error. The problem is that `rails-i18n` gem doesn't have
this key. It has `errors.template.{header.body}` instead. To fix this,
this fork https://github.com/duduribeiro/dynamic_form removes this
`activerecord` prefix from the i18n key and we uses this fork instead of
the dynamic_form gem.

A better way in the future is maybe removing this gem dependency from
the app.

After this PR:
image

### CODE REVIEW

**Comment by duduribeiro** [comment]
@dwradcliffe I think we can move this error messages creation into a helper. I think it can be easy. I can start working on this, what do you think?

**Comment by sonalkr132** [comment]
> we can move this error messages creation into a helper. 

All of code seem in single file. We can copy helpers we have used with your fix. :+1:

**Comment by duduribeiro** [comment]
@sonalkr132 and @dwradcliffe pushed the changes to this PR.

thanks so much 🍻

File: app/helpers/dynamic_errors_helper.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+module DynamicErrorsHelper
```
Comment (sonalkr132) [inline app/helpers/dynamic_errors_helper.rb]: If we are really adding this to app folder, we should compact it to exactly the code we use and add tests, which I think is better solution than copying to lib.

File: config/locales/pt-BR.yml [hunk 2]
```diff
@@ -23,6 +23,8 @@ pt-BR:
   try_again:
   advanced_search:
   activerecord:
+    models:
+      user: Usuário
```
Comment (sonalkr132) [inline config/locales/pt-BR.yml]: Please make a seperate PR with these.

File: Gemfile.lock [hunk 3]
```diff
@@ -462,4 +460,4 @@ DEPENDENCIES
   xml-simple
 
 BUNDLED WITH
-   1.16.2
+   1.16.5
```
Comment (sonalkr132) [inline Gemfile.lock]: Please don't change this.

**Comment by duduribeiro** [comment]
hey @sonalkr132 👋 

I did the changes suggested by you. thanks so much 🍺 

if needs anything more, just ping me

File: app/views/profiles/edit.html.erb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <% @title = t('.title') %>
 
 <%= form_for @user, :url => {:controller => 'profiles', :action => 'update'} do |form| %>
-  <%= form.error_messages %>
+  <%= error_messages_for(form.object) %>
```
Comment (sonalkr132) [inline app/views/profiles/edit.html.erb]: I think we need to pass `@user`?
```
ActionView::Template::Error: undefined method `errors' for :user:Symbol
    app/helpers/dynamic_errors_helper.rb:6:in `error_messages_for'
```

**Comment by sonalkr132** [comment]
Please fix rubocop warnings.

**Comment by duduribeiro** [comment]
@sonalkr132 fixed it.

thanks!

**Comment by sonalkr132** [comment]
Thanks @duduribeiro :tada:

**Pull request merged**